tests/unit/test_log.c

```diff
@@ -553,6 +555,103 @@ static void test_crc32c(void **state)
     assert_int_equal(sc_crc32(0, (uint8_t *) "1", 2), 2727214374);
 }
 
+/* Loop over log file header bytes and change one byte at a time.
+ * Verify we detect the corruption when we try to read the file. */
+static void test_corruption_header(void **state)
+{
+    /* Generate header on the disk. */
+    Log *log = LogCreate(LOGNAME, DBID, 1, 0, 1);
+    LogFree(log);
+
+    struct stat st;
+    stat(LOGNAME, &st);
+
+    /* Loop over the bytes and corrupt one byte each time. */
+    for (int i = 0; i < st.st_size; i++) {
+        int fd = open(LOGNAME, O_RDWR, S_IWUSR | S_IRUSR);
+        assert_true(fd > 0);
+
+        lseek(fd, i, SEEK_SET);
+        ssize_t rc = write(fd, "^", 1); /* Alter the byte */
+        (void) rc;
+        close(fd);
+
+        log = LogOpen(LOGNAME, 0);
+        assert_null(log);
+
+        /* Create file again. */
+        unlink(LOGNAME);
+        log = LogCreate(LOGNAME, DBID, 1, 0, 1);
+        LogFree(log);
+    }
+}
+
+/* Loop over an entry's bytes and change one byte at a time.
+ * Verify we detect the corruption when we try to read the file. */
+static void test_corruption_entry(void **state)
+{
+    raft_entry_t *e;
+
+    Log *log = LogCreate(LOGNAME, DBID, 1, 0, 1);
+    e = __make_entry_value(5000, "test5000");
+    LogAppend(log, e);
+    raft_entry_release(e);
+    LogFree(log);
+
+    struct stat st;
+    stat(LOGNAME, &st);
+
+    size_t entry_begin = st.st_size;
+
+    log = LogOpen(LOGNAME, 0);
+    LogLoadEntries(log);
+    e = __make_entry_value(6000, "test6000");
+    LogAppend(log, e);
+    raft_entry_release(e);
+    LogFree(log);
+
```

Comment:
suggest adding another entry in order to verify that the last entry (test7000) will be deleted too in corruption.

Reply:
done

Reply:
it should be added at the file init step too... 🙏

Reply:
ok, moved that part into the loop.

Also fixed a naming issue on the way, symbols starting with `__` reserved for C language/compilers. Linter was highlighting it.